zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1080,22 +1080,21 @@ protected void processStateChanged(WatchedEvent event) {
      */
     if (event.getState() == KeeperState.SyncConnected) {
       if (!_isNewSessionEventFired && !"0".equals(getHexSessionId())) {
-        if (isManagingZkConnection()) {
-          /*
-           * Before the new zookeeper instance is connected to the zookeeper service and its session
-           * is established, its session id is 0.
-           * New session event is not fired until the new zookeeper session receives the first
-           * SyncConnected state(the zookeeper session is established).
-           * Now the session id is available and non-zero, and we can fire new session events.
-           */
-          fireNewSessionEvents();
-        }
         /*
          * Set it true to avoid firing events again for the same session next time
          * when SyncConnected events are received.
          */
         _isNewSessionEventFired = true;
 
+        /*
+         * Before the new zookeeper instance is connected to the zookeeper service and its session
+         * is established, its session id is 0.
+         * New session event is not fired until the new zookeeper session receives the first
+         * SyncConnected state(the zookeeper session is established).
+         * Now the session id is available and non-zero, and we can fire new session events.
+         */
+        fireNewSessionEvents();
```

Comment:
Thanks for put the check inside the function. But why we need to move it to after _isNewSessionEventFired = true;?
Is it concerning? Because when _isNewSessionEventFired is set to true, the event is not sent yet.

Reply:
Any comments on "why we need to move it to after _isNewSessionEventFired = true;? Is it concerning?" ?

Reply:
I changed it back to the original way.